session/session.go

```diff
@@ -318,6 +327,10 @@ func (s *MenderShellSession) StartShell(sessionId string, terminal MenderShellTe
 	s.pseudoTTY = pseudoTTY
 	s.command = cmd
 	s.activeAt = timeNow()
+
+	// start the healthcheck go-routine
+	go s.healthcheck()
+
```

Comment:
I have some doubts if that go routine will be stopped in all cases when session is stopped. maybe we should make sure it is not left behind? (is it stopped on MenderSessionTerminateExpired?)

Reply:
I believe so, it is managed by `StopShell` which closes the `close` chan. This causes the goroutine to exit.
MenderSessionTerminateExpired is calling StopShell, indeed.